system/ui/lib/list_view.py

```diff
@@ -11,154 +9,266 @@
+# Type Aliases for Clarity
+StrSrc = str | Callable[[], str] | None
+BoolSrc = bool | Callable[[], bool]
 
-def _resolve_value(value, default=""):
+
+def _get_value(value, default=""):
   if callable(value):
     return value()
   return value if value is not None else default
 
 
-# Abstract base class for right-side items
-class ItemAction(Widget, ABC):
-  def __init__(self, width: int = 100, enabled: bool | Callable[[], bool] = True):
+class ListItem(Widget, ABC):
+  def __init__(self, title, description: StrSrc=None, enabled: BoolSrc=True, visible: BoolSrc=True, icon=None):
     super().__init__()
-    self.width = width
+    self.title = title
+    self._icon = icon
+    self.description = description
+    self.show_desc = False
+
     self._enabled_source = enabled
+    self._visible_source = visible
+    self._font = gui_app.font(FontWeight.NORMAL)
+
+    # Cached properties for performance
+    self._prev_max_width: int = 0
+    self._wrapped_description: str | None = None
+    self._prev_description: str | None = None
+    self._description_height: float = 0
 
   @property
   def enabled(self):
-    return _resolve_value(self._enabled_source, False)
+    return _get_value(self._enabled_source, True)
+
+  @property
+  def is_visible(self):
+    return _get_value(self._visible_source, True)
+
+  def set_enabled(self, enabled: bool):
+    self._enabled_source = enabled
+
+  def get_desc(self):
+    return _get_value(self.description, "")
+
+  def set_icon(self, icon: str):
+    self._icon = icon
+
+  def set_desc(self, description: StrSrc):
+    self.description = description
+    current_description = self.get_desc()
+    if current_description != self._prev_description:
+      self._update_description_cache(self._prev_max_width, current_description)
 
-  def get_width(self) -> int:
-    return self.width
+  def _update_description_cache(self, max_width: int, current_description: str):
+    """Update the cached description wrapping"""
+    self._prev_max_width = max_width
+    self._prev_description = current_description
+    content_width = max_width - ITEM_PADDING * 2
+
+    # Account for icon width
+    if self._icon:
+      content_width -= ICON_SIZE + ITEM_PADDING
+
+    wrapped_lines = wrap_text(self._font, current_description, ITEM_DESC_FONT_SIZE, content_width)
+    self._wrapped_description = "\n".join(wrapped_lines)
+    self._description_height = len(wrapped_lines) * ITEM_DESC_FONT_SIZE + 10
+
+  def _get_height(self, max_width: int) -> float:
+    if not self.is_visible:
+      return 0
+
+    if not self.show_desc:
+      return ITEM_BASE_HEIGHT
+
+    current_description = self.get_desc()
+    if not current_description:
+      return ITEM_BASE_HEIGHT
+
+    if current_description != self._prev_description or max_width != self._prev_max_width:
+      self._update_description_cache(max_width, current_description)
+
+    return ITEM_BASE_HEIGHT + self._description_height - (ITEM_BASE_HEIGHT - ITEM_DESC_V_OFFSET) + ITEM_PADDING
```

Comment:
```suggestion
  def _get_height(self, max_width: int) -> float:
    if not self.is_visible:
      return 0

    return self._rect
```

Does it make sense to modify the internal rect and return its cached height/width?

Reply:
The description can change dynamically, so the height may vary. The get_height method is responsible for updating the cache when a change is detected.

Reply:
Do we draw outside a widget's _rect?

Reply:
The ListViewItem is different from other widgets, so it needs to be handled separately